src/App.js

```diff
@@ -1,19 +1,131 @@
 import React from 'react';
+import { TodoList } from './components/TodoList';
 import './App.css';
+import todosFromApi from './api/todos';
+import usersFromApi from './api/users';
 
-import users from './api/users';
+function getUserById(UserId) {
+  return usersFromApi.find(user => user.id === UserId);
+}
+
+const preparedTodos = todosFromApi.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    users: usersFromApi,
+    taskTitle: '',
+    userId: 0,
+    errorUser: false,
+    errorTitle: false,
+  }
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+    const correctValue = name === 'userId' ? +value : value;
+
+    this.setState({ [name]: correctValue });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { userId, taskTitle } = this.state;
+
+    this.setState({
+      errorUser: false,
+      errorTitle: false,
+    });
+
+    if (userId === 0) {
+      return this.setState({ errorUser: true });
+    }
+
+    if (taskTitle.trim().length < 1) {
+      return this.setState({ errorTitle: true });
```

Comment:
how did you expect this should work? why do you return the result of setState? and return it to where?